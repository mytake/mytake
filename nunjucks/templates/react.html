{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
