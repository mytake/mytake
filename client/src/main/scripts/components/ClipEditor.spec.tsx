/*
 * MyTake.org website and tooling.
 * Copyright (C) 2018 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
import * as React from "react";
import * as renderer from "react-test-renderer";
import ClipEditor, { ClipEditorEventHandlers } from "./ClipEditor";
import {
  initialRangeSliders,
  unzoomedRangeSliders,
  zoomedRangeSliders,
  videoFactLink,
} from "../utils/testUtils";

jest.mock("./ZoomViewer");

jest.mock("./TrackSlider");

jest.mock("./QuickShare");

const eventHandlers: ClipEditorEventHandlers = {
  onAfterRangeChange: jest.fn(),
  onClearPress: jest.fn(),
  onPlayPausePress: jest.fn(),
  onRangeChange: jest.fn(),
  onRestartPress: jest.fn(),
  onSendToTake: jest.fn(),
  onSkipBackPress: jest.fn(),
  onSkipForwardPress: jest.fn(),
  onZoomToClipPress: jest.fn(),
};

test("Initial ClipEditor", () => {
  const tree = renderer
    .create(
      <ClipEditor
        eventHandlers={eventHandlers}
        captionIsHighlighted={false}
        currentTime={0}
        videoDuration={5224}
        isPaused={true}
        isZoomedToClip={false}
        rangeSliders={initialRangeSliders}
        stateAuthority={"SCROLL"}
        videoIdHash={videoFactLink.hash}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("ClipEditor with zoomed selection", () => {
  const tree = renderer
    .create(
      <ClipEditor
        eventHandlers={eventHandlers}
        captionIsHighlighted={true}
        currentTime={3.87}
        videoDuration={5224}
        isPaused={true}
        isZoomedToClip={true}
        rangeSliders={zoomedRangeSliders}
        stateAuthority={"SCROLL"}
        videoIdHash={videoFactLink.hash}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("ClipEditor with zoomed selection", () => {
  const tree = renderer
    .create(
      <ClipEditor
        eventHandlers={eventHandlers}
        captionIsHighlighted={true}
        currentTime={3.87}
        videoDuration={5224}
        isPaused={true}
        isZoomedToClip={false}
        rangeSliders={unzoomedRangeSliders}
        stateAuthority={"SCROLL"}
        videoIdHash={videoFactLink.hash}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
