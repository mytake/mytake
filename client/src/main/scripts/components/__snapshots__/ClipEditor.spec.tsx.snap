// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClipEditor with zoomed selection 1`] = `
<div
  className="clipEditor"
>
  <div
    className="clipEditor__actions clipEditor__actions--range"
  >
    <TrackSlider
      end={5224}
      eventHandlers={
        Object {
          "onAfterRangeChange": [MockFunction],
          "onRangeChange": [Function],
        }
      }
      rangeSliders={
        Array [
          Object {
            "label": "Now playing",
            "start": 3.87,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#d3dae3",
              },
            },
            "type": "CURRENT_TIME",
          },
          Object {
            "end": 28.17,
            "label": "Transcript",
            "start": 3.659,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#758aa8",
              },
            },
            "type": "VIEW",
          },
          Object {
            "end": 5.223,
            "label": "Zoom",
            "start": 3.7470000000000003,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#758aa8",
              },
            },
            "type": "ZOOM",
          },
        ]
      }
      start={0}
      stateAuthority="SCROLL"
      step={1}
    />
  </div>
  <ZoomViewer
    duration={5224}
    zoomRange={
      Object {
        "end": 5.223,
        "label": "Zoom",
        "start": 3.7470000000000003,
        "styles": Object {
          "handle": Object {
            "backgroundColor": "#758aa8",
            "border": "1px solid #2c4770",
          },
          "rail": Object {
            "backgroundColor": "#d3dae3",
          },
          "track": Object {
            "backgroundColor": "#758aa8",
          },
        },
        "type": "ZOOM",
      }
    }
  />
  <div
    className="clipEditor__actions clipEditor__actions--range"
  >
    <TrackSlider
      end={5.223}
      eventHandlers={
        Object {
          "onAfterRangeChange": [MockFunction],
          "onRangeChange": [Function],
        }
      }
      rangeSliders={
        Array [
          Object {
            "label": "Now playing",
            "start": 3.87,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#d3dae3",
              },
            },
            "type": "CURRENT_TIME",
          },
          Object {
            "end": 28.17,
            "label": "Transcript",
            "start": 3.659,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#758aa8",
              },
            },
            "type": "VIEW",
          },
          Object {
            "end": 5.1,
            "label": "Clip",
            "start": 3.87,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#758aa8",
              },
            },
            "type": "SELECTION",
          },
        ]
      }
      start={3.7470000000000003}
      stateAuthority="SCROLL"
      step={0.1}
    />
  </div>
  <div
    className="clipEditor__actions clipEditor__actions--controls"
  >
    <div
      className="clipEditor__controls"
    >
      <button
        className="clipEditor__button clipEditor__button--zoom"
        onClick={[Function]}
      >
        Play Clip
      </button>
      <button
        className="clipEditor__button clipEditor__button--small"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-undo"
        />
         15s
      </button>
      <button
        className="clipEditor__button clipEditor__button--small"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-play"
        />
      </button>
      <button
        className="clipEditor__button clipEditor__button--small"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-repeat"
        />
         15s
      </button>
      <div
        className="dropdown dropdown--clipShare"
      >
        <span
          className="dropdown__toggle dropdown__toggle--clipShare"
          onClick={[Function]}
        >
          Share
        </span>
        <div
          className="dropdown__dropdown dropdown__dropdown--hidden"
        >
          <QuickShare
            factHash="iEfwIxM7MmnOKb7zt4HqW8IxUWy6F7a236fSOQlUUWI="
            highlightedRange={
              Array [
                3.87,
                5.1,
              ]
            }
            isDocument={false}
            onSendToTake={[MockFunction]}
            viewRange={
              Array [
                3.7470000000000003,
                5.223,
              ]
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClipEditor with zoomed selection 2`] = `
<div
  className="clipEditor"
>
  <div
    className="clipEditor__actions clipEditor__actions--range"
  >
    <TrackSlider
      end={5224}
      eventHandlers={
        Object {
          "onAfterRangeChange": [MockFunction],
          "onRangeChange": [Function],
        }
      }
      rangeSliders={
        Array [
          Object {
            "label": "Now playing",
            "start": 3.87,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#d3dae3",
              },
            },
            "type": "CURRENT_TIME",
          },
          Object {
            "end": 28.17,
            "label": "Transcript",
            "start": 3.659,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#758aa8",
              },
            },
            "type": "VIEW",
          },
          Object {
            "end": 82,
            "label": "Zoom",
            "start": 4,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#758aa8",
              },
            },
            "type": "ZOOM",
          },
        ]
      }
      start={0}
      stateAuthority="SCROLL"
      step={1}
    />
  </div>
  <ZoomViewer
    duration={5224}
    zoomRange={
      Object {
        "end": 82,
        "label": "Zoom",
        "start": 4,
        "styles": Object {
          "handle": Object {
            "backgroundColor": "#758aa8",
            "border": "1px solid #2c4770",
          },
          "rail": Object {
            "backgroundColor": "#d3dae3",
          },
          "track": Object {
            "backgroundColor": "#758aa8",
          },
        },
        "type": "ZOOM",
      }
    }
  />
  <div
    className="clipEditor__actions clipEditor__actions--range"
  >
    <TrackSlider
      end={82}
      eventHandlers={
        Object {
          "onAfterRangeChange": [MockFunction],
          "onRangeChange": [Function],
        }
      }
      rangeSliders={
        Array [
          Object {
            "label": "Now playing",
            "start": 3.87,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#d3dae3",
              },
            },
            "type": "CURRENT_TIME",
          },
          Object {
            "end": 28.17,
            "label": "Transcript",
            "start": 3.659,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#758aa8",
              },
            },
            "type": "VIEW",
          },
          Object {
            "end": 5.1,
            "label": "Clip",
            "start": 3.87,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#758aa8",
              },
            },
            "type": "SELECTION",
          },
        ]
      }
      start={4}
      stateAuthority="SCROLL"
      step={0.1}
    />
  </div>
  <div
    className="clipEditor__actions clipEditor__actions--controls"
  >
    <div
      className="clipEditor__controls"
    >
      <button
        className="clipEditor__button clipEditor__button--zoom"
        onClick={[MockFunction]}
      >
        Zoom to Clip
      </button>
      <button
        className="clipEditor__button clipEditor__button--small"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-undo"
        />
         15s
      </button>
      <button
        className="clipEditor__button clipEditor__button--small"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-play"
        />
      </button>
      <button
        className="clipEditor__button clipEditor__button--small"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-repeat"
        />
         15s
      </button>
      <div
        className="dropdown dropdown--clipShare"
      >
        <span
          className="dropdown__toggle dropdown__toggle--clipShare"
          onClick={[Function]}
        >
          Share
        </span>
        <div
          className="dropdown__dropdown dropdown__dropdown--hidden"
        >
          <QuickShare
            factHash="iEfwIxM7MmnOKb7zt4HqW8IxUWy6F7a236fSOQlUUWI="
            highlightedRange={
              Array [
                3.87,
                5.1,
              ]
            }
            isDocument={false}
            onSendToTake={[MockFunction]}
            viewRange={
              Array [
                4,
                82,
              ]
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Initial ClipEditor 1`] = `
<div
  className="clipEditor"
>
  <div
    className="clipEditor__actions clipEditor__actions--range"
  >
    <TrackSlider
      end={5224}
      eventHandlers={
        Object {
          "onAfterRangeChange": [MockFunction],
          "onRangeChange": [Function],
        }
      }
      rangeSliders={
        Array [
          Object {
            "label": "Now playing",
            "start": 0,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#d3dae3",
              },
            },
            "type": "CURRENT_TIME",
          },
          Object {
            "end": 18.119,
            "label": "Transcript",
            "start": 0.75,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#758aa8",
              },
            },
            "type": "VIEW",
          },
          Object {
            "end": 5224,
            "label": "Zoom",
            "start": 0,
            "styles": Object {
              "handle": Object {
                "backgroundColor": "#758aa8",
                "border": "1px solid #2c4770",
              },
              "rail": Object {
                "backgroundColor": "#d3dae3",
              },
              "track": Object {
                "backgroundColor": "#758aa8",
              },
            },
            "type": "ZOOM",
          },
        ]
      }
      start={0}
      stateAuthority="SCROLL"
      step={1}
    />
  </div>
  <div
    className="clipEditor__actions clipEditor__actions--range"
  >
    <p
      className="clipEditor__text clipEditor__text--range"
    >
      Drag the range handles to create a clip
    </p>
  </div>
  <div
    className="clipEditor__actions clipEditor__actions--controls"
  >
    <div
      className="clipEditor__controls"
    >
      <div
        className="clipEditor__spacer"
      />
      <button
        className="clipEditor__button clipEditor__button--small"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-undo"
        />
         15s
      </button>
      <button
        className="clipEditor__button clipEditor__button--small"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-play"
        />
      </button>
      <button
        className="clipEditor__button clipEditor__button--small"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-repeat"
        />
         15s
      </button>
    </div>
  </div>
</div>
`;
